### PR TITLE
Worked on refactoring project structure

### PR DESCRIPTION
**Pull request by KritikHub**
- Added MovieViewModel, MovieListData, Views folder for the Home module
- Refactored MovieListView file
- Modified URN file

### CODE REVIEW

**Comment by KritikHub** [comment]
@JustManish Please review the code.

File: SwiftUiMovieDb/Bindings/MovieDetailState.swift [hunk 1]
```diff
@@ -10,7 +10,7 @@ import SwiftUI
 class MovieDetailState: ObservableObject {
     
     private let movieService: MovieService
-    @Published var movie: Movie?
```
Comment (JustManish) [inline SwiftUiMovieDb/Bindings/MovieDetailState.swift:13]: Why this property is optional ?

File: SwiftUiMovieDb/Bindings/MovieDetailState.swift [hunk 2]
```diff
@@ -10,7 +10,7 @@ import SwiftUI
...
+    @Published var movie: Movies?
     @Published var isLoading = false
     @Published var error: NSError?
```
Comment (JustManish) [inline SwiftUiMovieDb/Bindings/MovieDetailState.swift:15]: Why NSError ? it should be more Project specific Error likeMovieDBError Type ?

File: SwiftUiMovieDb/Bindings/MovieListState.swift [hunk 3]
```diff
@@ -9,7 +9,7 @@ import SwiftUI
 
 class MovieListState: ObservableObject {
     
-    @Published var movies: [Movie]?
```
Comment (JustManish) [inline SwiftUiMovieDb/Bindings/MovieListState.swift:12]: why optional ? just use an empty array instead.
eg: @Published var movies: [Movie] = []

File: SwiftUiMovieDb/Bindings/MovieSearchState.swift [hunk 4]
```diff
@@ -12,7 +12,7 @@ import Foundation
 class MovieSearchState: ObservableObject {
     
     @Published var query = ""
-    @Published var movies: [Movie]?
+    @Published var movies: [Movies]?
     @Published var isLoading = false
     @Published var error: NSError?
```
Comment (JustManish) [inline SwiftUiMovieDb/Bindings/MovieSearchState.swift:17]: Again don't use NSError.

File: SwiftUiMovieDb/Models/Movie+Stub.swift [hunk 5]
```diff
@@ -7,13 +7,13 @@
 
 import Foundation
 
-extension Movie {
-    static var stubbedMovies: [Movie] {
+extension Movies {
+    static var stubbedMovies: [Movies] {
         let response: MovieResponse? = try? Bundle.main.loadAndDecodeJSON(filename: "movie_list")
         return response!.results
```
Comment (JustManish) [inline SwiftUiMovieDb/Models/Movie+Stub.swift:13]: Avoid force unwraps instead handle optional's.

File: SwiftUiMovieDb/Models/Movie+Stub.swift [hunk 6]
```diff
@@ -7,13 +7,13 @@
...
     }
     
-    static var stubbedMovie: Movie {
+    static var stubbedMovie: Movies {
         stubbedMovies[0]
```
Comment (JustManish) [inline SwiftUiMovieDb/Models/Movie+Stub.swift:17]: use stubedMovies.first instead.

File: SwiftUiMovieDb/Modules/Home/MovieViewModel/MovieListViewModel.swift [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    
+    let service = APIService()
+    
+   func loadMovies(with movieType: MovieListType) {
+        self.movies = nil
+        self.isLoading = false
+    let urn = MovieListURN(movieType: movieType)
+        self.service.makeRequest(with: urn) {[weak self] (result) in
+            guard let self = self else { return }
+            DispatchQueue.main.async {
```
Comment (JustManish) [inline SwiftUiMovieDb/Modules/Home/MovieViewModel/MovieListViewModel.swift:24]: Reindent code here

File: SwiftUiMovieDb/Modules/Home/Views/MovieBackdropCard.swift [hunk 8]
```diff
@@ -36,14 +36,13 @@ struct MovieBackdropCard: View {
...
     Image("PosterImagePlaceholder")
         .resizable()
 }
 
-struct MovieBackdropCard_Previews: PreviewProvider {
-    static var previews: some View {
-        MovieBackdropCard(movie: Movie.stubbedMovie)
-    }
-}
+//struct MovieBackdropCard_Previews: PreviewProvider {
```
Comment (JustManish) [inline SwiftUiMovieDb/Modules/Home/Views/MovieBackdropCard.swift:44]: Remove if preview not needed

File: SwiftUiMovieDb/Views/LoadingMDBView.swift [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+                HStack {
+                    Spacer()
+                    ActivityIndicatorView()
+                    Spacer()
+                }
+            } else if error != nil {
+                HStack {
+                    Spacer()
+                    VStack(spacing: 4) {
+                        Text(error!.localizedDescription).font(.headline)
```
Comment (JustManish) [inline SwiftUiMovieDb/Views/LoadingMDBView.swift:28]: don't use force unwrap rather handle it.

File: SwiftUiMovieDb/Views/LoadingMDBView.swift [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+                        if self.retryAction != nil {
```
Comment (JustManish) [inline SwiftUiMovieDb/Views/LoadingMDBView.swift:29]: use if let instead here.

File: SwiftUiMovieDb/Views/LoadingMDBView.swift [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+    
+    var body: some View {
+        Group {
+            if isLoading {
+                HStack {
+                    Spacer()
+                    ActivityIndicatorView()
+                    Spacer()
+                }
+            } else if error != nil {
```
Comment (JustManish) [inline SwiftUiMovieDb/Views/LoadingMDBView.swift:24]: use if let here

**Review by JustManish — CHANGES_REQUESTED** [review]

**Pull request closed without merging**